### PR TITLE
Add coveo-eks module

### PR DESCRIPTION
**Pull request by tdegiacinto**

### CODE REVIEW

File: coveo-eks/_locals.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+locals {
+  kiam_worker_groups = [{
```
Comment (jmorissette) [inline coveo-eks/_locals.tf]: Wouldnt we want all those parameter to be configurable by an user of the module ?

File: coveo-eks/_locals.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  workers_group_defaults_overrides = {
+    default = {}
+
+    ubuntu = "${map("ami_id", data.aws_ami.ubuntu.image_id)}" // Here it imply that an ami for ubuntu is already available before creating an EKS cluster
+  }
+
+  worker_group_kiam_tags = {
+    kiam = [{
+      key                 = "hostname-prefix"
+      value               = "${var.prefix}-eks"
```
Comment (jmorissette) [inline coveo-eks/_locals.tf]: Has it was intended to be open source we tried not to used var magically included by tgf :/

File: coveo-eks/_outputs.tf [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+output "worker_iam_role_arn" {
+  description = "default IAM role ARN for EKS worker groups"
+  value       = "${module.k8s-cluster.worker_iam_role_arn}"
+}
+
+output "cluster_namespaces" {
+  description = "All namespaces created in the cluster"
+  value       = "${concat(list("infrastructure", "csec" , "monitoring", "ingress"), var.namespaces)}"
```
Comment (jmorissette) [inline coveo-eks/_outputs.tf]: Shouldnt the list be all configurable ?

File: coveo-eks/_locals.tf (hunk 1 above)
Comment (tdegiacinto) [reply coveo-eks/_locals.tf]: No we don't.

File: coveo-eks/_locals.tf (hunk 2 above)
Comment (tdegiacinto) [reply coveo-eks/_locals.tf]: Mmmm I may be wrong but opensource stuff are in coveooss no ? We don't want to opensource this module.

File: coveo-eks/_outputs.tf (hunk 3 above)
Comment (tdegiacinto) [reply coveo-eks/_outputs.tf]: it is , we have some default namespaces (infra,csec,...) merged with the variable `namespaces` which contain all ns defined by user.

**Pull request merged**